src/core/distributions/phylogenetics/tree/birthdeath/BirthDeathSamplingTreatmentProcess.cpp

```diff
@@ -1590,36 +1594,52 @@ double BirthDeathSamplingTreatmentProcess::simulateDivergenceTime(double origin,
  * \return The diversity (number of species in the reconstructed tree).
  */
 int BirthDeathSamplingTreatmentProcess::survivors(double t) const
-{
-
-    const std::vector<TopologyNode*>& nodes = value->getNodes();
-
+{  
+    
     int survivors = 0;
-    for (std::vector<TopologyNode*>::const_iterator it = nodes.begin(); it != nodes.end(); ++it)
+
+    if ( use_origin )
     {
-        TopologyNode* n = *it;
-        double my_age = n->getAge();
+        if ( t > getOriginAge() )
+        {
+            return 0;
+        }
+        survivors = 1;
+    } else {
+        if ( t > value->getRoot().getAge() )
+        {
+            return 0;
+        }
+        survivors = 2;
+    }
 
-        // my age needs to be smaller that the requested time
-        if ( (my_age - t) < 1E-4 )
+    for (size_t i=0; i<serial_bifurcation_times.size(); ++i) {
+        if (t < serial_bifurcation_times[i])
```

Comment:
I'm not sure if this checking for the specific time is safe, because our io usually writes branch lengths rounded to 4 decimals.

Reply:
@hoehna We've already addressed the classification-time issue elsewhere though. countAllNodes() is always called first, which uses whichIntervalTime() to classify node ages, and _that_ function uses the 1e-4 tolerance

Reply:
In other words, we outsource to other functions the matter of "is this (potentially) low-precision tree time equal to this high-precision event-time" and so now we can just compare doubles to doubles, give or take possible a <=

Reply:
maybe add a note here about classification & tolerances being handled in countAllNodes / whichIntervalTime?